tests/receivers/mysql/mysql_test.go

```diff
@@ -0,0 +1,45 @@
+// Copyright Splunk, Inc.
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build integration
+
+package tests
+
+import (
+	"path"
+	"testing"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+var mysql = []testutils.Container{testutils.NewContainer().WithContext(
+	path.Join(".", "testdata", "server"),
+).WithName("mysql").WithExposedPorts("3306:3306").WillWaitForPorts("3306").WillWaitForLogs("X Plugin ready for connections. Bind-address:")}
+
+// This test ensures the collector can connect to a MySQL DB, and properly get metrics. It's not intended to
+// test the receiver itself.
```

Comment:
wading into semantics, these tests are partially intended to test the underlying components. Though some of the expectations we have about the reported telemetry may need to be updated (resource metric file changes), they can detect breaking changes or bugs that may have been missed upstream.